projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/PersonalDetailsService.kt

```diff
@@ -0,0 +1,106 @@
+package uk.gov.justice.digital.hmpps.service
+
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody
+import uk.gov.justice.digital.hmpps.alfresco.AlfrescoClient
+import uk.gov.justice.digital.hmpps.api.model.Name
+import uk.gov.justice.digital.hmpps.api.model.overview.PersonalCircumstance
+import uk.gov.justice.digital.hmpps.api.model.personalDetails.*
+import uk.gov.justice.digital.hmpps.integrations.delius.overview.entity.Person
+import uk.gov.justice.digital.hmpps.integrations.delius.personalDetails.entity.*
+import uk.gov.justice.digital.hmpps.integrations.delius.personalDetails.entity.ContactAddress
+import uk.gov.justice.digital.hmpps.integrations.delius.personalDetails.entity.PersonalContact
+
+@Service
+class PersonalDetailsService(
+    private val personalDetailsRepository: PersonalDetailsRepository,
+    private val addressRepository: PersonAddressRepository,
+    private val documentRepository: DocumentRepository,
+    private val alfrescoClient: AlfrescoClient
+) {
+
+    @Transactional
+    fun getPersonalDetails(crn: String): PersonalDetails {
+        val person = personalDetailsRepository.getPersonDetails(crn)
+        val addresses = addressRepository.findByPersonId(person.id)
+        val mainAddress = addresses.firstOrNull { it.status.code == "M" }?.toAddress()
+        val otherAddresses = addresses.filter { it.status.code != "M" }.map(PersonAddress::toAddress).mapNotNull { it }
+        val documents = documentRepository.findByPersonId(person.id)
+
+        return PersonalDetails(
+            crn = person.crn,
+            name = person.name(),
+            mainAddress = mainAddress,
+            otherAddresses = otherAddresses,
+            contacts = person.personalContacts.map(PersonalContact::toContact),
+            preferredGender = person.gender.description,
+            dateOfBirth = person.dateOfBirth,
+            preferredName = person.preferredName,
+            telephoneNumber = person.telephoneNumber,
+            mobileNumber = person.mobileNumber,
+            circumstances = Circumstances(lastUpdated = person.personalCircumstances.maxByOrNull { it.lastUpdated }?.lastUpdated,
+                circumstances = person.personalCircumstances.map {
+                    PersonalCircumstance(
+                        it.subType.description,
+                        it.type.description
+                    )
+                }),
+            disabilities = Disabilities(
+                lastUpdated = person.disabilities.maxByOrNull { it.lastUpdated }?.lastUpdated,
```

Comment:
Minor thing - you should be able to replace thes `maxByOrNull`s with `maxOfOrNull`
```suggestion
                lastUpdated = person.disabilities.maxOfOrNull { it.lastUpdated },
```

Reply:
Done